rust_src/src/buffers.rs

```diff
@@ -1843,4 +1847,80 @@ pub fn byte_char_debug_check(b: LispBufferRef, charpos: isize, bytepos: isize) {
     }
 }
 
+// True if B can be used as 'other-than-BUFFER' buffer.
+fn candidate_buffer(b: LispObject, buffer: LispObject) -> bool {
```

Comment:
Can this function replace the C function?

Reply:
Yes. I've just `pub extern "C"`'d it.

Reply:
Or rather, it was only used by other_buffer and other_buffer_safely, and I've made the latter visible to C, so the C implementation of both of those can go.